design-book/src/design-constraints.md

```diff
@@ -0,0 +1,58 @@
+  - This keeps our repository clean, and allows users to steal our work to use in their own projects and tooling.
+- Widgets, styles, colors and interaction paradigms should be consistent across the editor.
+  - This avoids duplication, makes changes easier and helps give the editor a cohesive look.
+- The editor should be distributed as an executable, either directly from the Bevy website or via Steam.
+  - This makes it easier for non-technical users to install.
+- The editor should be configurable, and easily adapted to both different genres and specific games.
+  - Every game needs a different set of features. We need to allow teams to expose or extend what they care about, and hide what they don't.
+- Industry-standard conventions for controls, UX and so on should be followed unless there's a good reason not to.
+  - Learning a new tool is hard enough: please don't make it any harder.
+- The Bevy editor should be stable, and recover from crashes in the game or the editor as gracefully as possible.
+  - Losing work sucks a lot!
+- Iteration times should be kept fast, using whatever means make sense for a given workflow.
+  - Compiling Rust is slow, and more iterations speeds up work and leads to a better result.
+- Version-controllable: assets and settings should be feasible to check into version control and be relatively robust to merge conflicts.
+  - This makes sharing settings across teams easier, and reduces confusing and wasteful problems with consolodating changes.
+- Integrate nicely with other related Bevy initiatives, namely dev tools and the Bevy CLI.
+  - We should avoid duplicating work and stepping on each others toes.
+
+## Should not
+
+- 3rd party Bevy dependencies should be used sparingly, and only with the consent of the maintainer.
+  - Bevy's ecosystem is one of its strengths, and we should lean on it, but being incorporated into the editor is a huge maintenance burden.
+- The editor is not an asset or code creation tool (certainly not in the default configuration). If an established tool does a good job authoring content, focus on importing its output, not reinventing Blender, VSCode, Aseprite, and Reaper.
+  - We have limited resources, and this isn't a good use of them. Artists and professionals already know and prefer specialized tools.
+- The Bevy editor should not have a native look-and-feel, and should not use native menus.
+  - This is a poor use of resources, leads to inconsistent aesthetics, and makes it harder to test and teach across platforms. Instead, focus on the useful behavior and design conventions.
+  - File picking widgets are an exception: these are non-intrusive, good crates exist and are not central to the user experience.
+- End users should not have to install or update Rust.
+  - This is a relatively challenging task for non-technical users, and should be handled transparently by the installer if it's absolutely needed.
```

Comment:
This feels like something that can be achieved once we reach a point that the API is mature and stable, with little to no user migrations. Only new features and underlying APIs